screens/OnboardingScreen.js

```diff
@@ -0,0 +1,113 @@
+import { t } from 'react-native-tailwindcss';
+import { Modal, Portal, Text, TextInput, HelperText } from 'react-native-paper';
+import WelcomeSwiper from '../components/Swiper/WelcomeSwiper';
+import StyledButton from '../components/StyledButton';
+
+export default function OnboardingScreen() {
+  // Manage login states
+  const [loginModalVisible, setloginModalVisibility] = useState(false);
+  const [loginUsername, setLoginUsername] = useState('');
+  const [loginPassword, setLoginPassword] = useState('');
+  const showLoginModal = () => setloginModalVisibility(true);
+  const hideLoginModal = () => setloginModalVisibility(false);
+
+  // Manage signup states
+  const [signUpModalVisible, setSignUpModalVisibility] = useState(false);
+  const [signUpUsername, setSignUpUsername] = useState('');
+  const [signUpPassword, setSignUpPassword] = useState('');
+  const [signUpEmail, setSignUpEmail] = useState('');
+  const showSignUpModal = () => setSignUpModalVisibility(true);
+  const hideSignUpModal = () => setSignUpModalVisibility(false);
+  const handleCreateAnAccount = () => {
+    hideLoginModal();
+    showSignUpModal();
+  };
+  const hasErrors = (email) => {
+    return !email.includes('@');
+  };
+
+  return (
+    <View style={[t.flex1, t.justifyCenter, t.itemsCenter, t.bgGreen600]}>
+      {/* Onboarding swiper slider */}
+      <WelcomeSwiper />
+      {/* Login Modal */}
+      <Portal>
+        <Modal
+          contentContainerStyle={[t.bgWhite, t.p8]}
+          visible={loginModalVisible}
+          onDismiss={hideLoginModal}
+        >
+          <TextInput
+            label="Username"
+            value={loginUsername}
+            onChangeText={(text) => setLoginUsername(text)}
+          />
+          <TextInput
+            label="Password"
+            value={loginPassword}
+            onChangeText={(text) => setLoginPassword(text)}
+          />
+          <Text style={[t.textRight, t.pT1, t.pB1]} onPress={handleForgotPasswordPress}>
+            Forgot Password?
+          </Text>
+          <StyledButton title="Login" mode="outlined" size="small" onPress={handleLoginPress} />
+          <Text style={[t.textCenter, t.pTPx]} onPress={handleCreateAnAccount}>
+            Create An Account
+          </Text>
+        </Modal>
+      </Portal>
+      {/* Sign Up Modal */}
+      <Portal>
+        <Modal
+          contentContainerStyle={[t.bgWhite, t.p8]}
+          visible={signUpModalVisible}
+          onDismiss={hideSignUpModal}
+        >
+          <TextInput
+            label="Username"
+            value={signUpUsername}
+            onChangeText={(text) => setSignUpUsername(text)}
+          />
+          <TextInput
+            label="Password"
+            value={signUpPassword}
+            onChangeText={(text) => setSignUpPassword(text)}
+          />
+          <TextInput
+            label="Email"
+            value={signUpEmail}
+            onChangeText={(text) => setSignUpEmail(text)}
+          />
```

Comment:
https://reactnative.dev/docs/textinput

Reply:
Good catch! I added the secureTextEntry for password input in SignUp and Login modals.